modules/artifact/src/test/java/org/mule/runtime/module/artifact/classloader/IBMMQResourceReleaserPropertiesTriggerTestCase.java

```diff
@@ -0,0 +1,160 @@
+      public ClassLoaderLookupPolicy extend(Map<String, LookupStrategy> lookupStrategies) {
+        return null;
+      }
+
+      @Override
+      public ClassLoaderLookupPolicy extend(Map<String, LookupStrategy> lookupStrategies, boolean overwrite) {
+        return null;
+      }
+    };
+  }
+
+
+  @Parameterized.Parameters(name = "Testing Driver {0}")
+  public static String[] data() throws NoSuchFieldException, IllegalAccessException {
+    return new String[] {
+        "9.2.5.0",
+        "9.2.4.0",
+        "9.2.3.0",
+        "9.2.2.0",
+        "9.1.1.0"
+    };
+  }
+
+  @Before
+  public void setup() throws Exception {
+    Properties props = System.getProperties();
+    props.setProperty("avoid.ibm.mq.cleanup.mbeans", "true");
+    URL settingsUrl = getClass().getClassLoader().getResource("custom-settings.xml");
+    final MavenClientProvider mavenClientProvider = discoverProvider(this.getClass().getClassLoader());
+
+    final Supplier<File> localMavenRepository =
+        mavenClientProvider.getLocalRepositorySuppliers().environmentMavenRepositorySupplier();
+
+    final MavenConfiguration.MavenConfigurationBuilder mavenConfigurationBuilder =
+        newMavenConfigurationBuilder().globalSettingsLocation(toFile(settingsUrl));
+
+    MavenClient mavenClient = mavenClientProvider
+        .createMavenClient(mavenConfigurationBuilder.localMavenRepositoryLocation(localMavenRepository.get()).build());
+
+    BundleDescriptor bundleDescriptor = new BundleDescriptor.Builder().setGroupId(DRIVER_GROUP_ID)
+        .setArtifactId(DRIVER_ARTIFACT_ID).setVersion(driverVersion).build();
+
+    dependency = mavenClient.resolveBundleDescriptor(bundleDescriptor);
+
+    artifactClassLoader =
+        new MuleArtifactClassLoader("IBMMQResourceReleaserPropertiesTriggerTestCase", mock(ArtifactDescriptor.class),
+                                    new URL[] {dependency.getBundleUri().toURL()},
+                                    currentThread().getContextClassLoader(), testLookupPolicy);
+  }
+
+  @Test
+  @Description("When redeploying an application which contains the IBM MQ Driver, the proper cleanup should be performed " +
+      "on redeployment but, if the property avoid.ibm.mq.cleanup.mbeans=true, the mbeans clean should be skiped.")
+  public void releaserMBeansPropertyTrueTriggerTest() throws Exception {
+    Field shouldReleaseIbmMQResourcesField = getField(MuleArtifactClassLoader.class, "shouldReleaseIbmMQResources", false);
+    shouldReleaseIbmMQResourcesField.setAccessible(true);
+    assertThat(shouldReleaseIbmMQResourcesField.get(artifactClassLoader), is(false));
+    // Force to load a Driver class so the resource releaser is flagged to run on dispose
+    Class<?> connectionFactoryClass = Class.forName(KNOWN_DRIVER_CLASS_NAME, true, artifactClassLoader);
+    Object connectionFactory = connectionFactoryClass.newInstance();
+    artifactClassLoader.dispose();
+    assertThat(countMBeans(artifactClassLoader), is(2));
```

Comment:
you can use the countMBeans at the beginning of the test and assign it to a variable to replace the 2 here.